internal/controllers/reference_addon_controller.go

```diff
@@ -0,0 +1,61 @@
+package controllers
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	refapisv1alpha1 "github.com/openshift/reference-addon/apis/reference/v1alpha1"
+)
+
+type ReferenceAddonReconciler struct {
+	client.Client
+	Log             logr.Logger
+	Scheme          *runtime.Scheme
+	latestHeartbeat metav1.Condition
```

Comment:
Don't we want to guard `latestHeartbeat` using a mutex, since the `ReferenceAddonReconciler` object is shared across go-routines (on each reconcile invocation)?

Reply:
https://github.com/openshift/reference-addon/pull/5#issuecomment-986844956 :)

Reply:
Sorry my bad!